internal/storage/persistence/ent/client/client.go

```diff
@@ -223,27 +229,27 @@ func (c *Client) UpdateAIP(
 	// Apply the updater function to the loaded AIP.
 	up, err := updater(convertDBAIP(dbAIP))
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, rollback(tx, err)
 	}
 
 	// Build a query to update the AIP in the database.
-	q := c.c.AIP.UpdateOneID(dbAIP.ID)
+	q := tx.AIP.UpdateOneID(dbAIP.ID)
 
 	if up.Status != "" && up.Status != dbAIP.Status {
 		q.SetStatus(up.Status)
 		updated = true
 	}
 
-	if up.LocationUUID != nil && up.LocationUUID.String() != locUUID.String() {
+	if !uuidsEqual(up.LocationUUID, locUUID) {
 		loc, err := c.c.Location.Query().
```

Comment:
Should we use the tx handle?

```suggestion
		loc, err := tx.Location.Query().
```

Reply:
Oops, missed that one. :smile: